TypeSystem/Schema/smoke_test_struct.h

```diff
@@ -68,6 +68,16 @@ CURRENT_STRUCT_T(TemplatedInheriting, A) {
   CURRENT_FIELD(meh, T);
 };
 
+CURRENT_STRUCT(TrickySchemaCases) {
```

Comment:
s/TrickySchemaCases/TrickyOptionalCases/ and move `w1`-`w6` below to `TrickyTemplatedCases`?